src/server/api/routers/traces.ts

```diff
@@ -360,4 +360,51 @@ export const traceRouter = createTRPCRouter({
         throw e;
       }
     }),
+  deleteMany: protectedProjectProcedure
+    .input(
+      z.object({
+        traceIds: z.array(z.string()),
+        projectId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      throwIfNoAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "traces:delete",
+      });
+
+      if (input.traceIds.length < 1) {
+        throw new Error("Minimum 1 trace_Id is required.");
+      }
+
+      const traces = await ctx.prisma.trace.findMany({
```

Comment:
I'd suggest to directly delete without first checking if they exist. This would only fail anyway if no trace was found. If some are missing this if(!traces) check does not work